src/pages/HomePage.jsx

```diff
@@ -1,17 +1,30 @@
-import { Box, Stack, Typography } from '@mui/material';
+import { Box, Stack, Typography, Button } from '@mui/material';
 import { Home } from '@mui/icons-material';
 import { useEffect } from 'react';
 import { useDispatch, useSelector } from 'react-redux';
 import { selectMyData } from '../store/sampleFeature/selectors';
 import { getMyData } from '../store/sampleFeature/slice';
+import { useNavigate } from 'react-router-dom';
 
 const HomePage = () => {
+  const navigate = useNavigate();
   const dispatch = useDispatch();
   const myData = useSelector(selectMyData);
+  const authToken = localStorage.getItem('authToken');
```

Comment:
create a helper function named **isLoggedIn** in the file named **auth.js** to handle this logic